src/build_queue.rs

```diff
@@ -282,29 +310,24 @@ impl BuildQueue {
 
         for change in &changes {
             match change {
-                Change::Yanked(release) => {
-                    let res = conn
-                        .execute(
-                            "
-                            UPDATE releases
-                                SET yanked = TRUE
-                            FROM crates
-                            WHERE crates.id = releases.crate_id
-                                AND name = $1
-                                AND version = $2
-                            ",
-                            &[&release.name, &release.version],
-                        )
-                        .with_context(|| {
-                            format!(
-                                "error while setting {}-{} to yanked",
-                                release.name, release.version
-                            )
-                        });
-                    match res {
-                        Ok(_) => debug!("{}-{} yanked", release.name, release.version),
-                        Err(err) => report_error(&err),
-                    }
+                Change::Yanked(release)
+                | Change::Unyanked(release)
+                // FIXME: the correct handling of `AddedAndYanked` is probably 
+                // running the build and direcctly yanking the release. 
+                // looking at 
+                // https://github.com/Byron/crates-index-diff-rs/commit/8256cbbd3651073394d6aa9de38c734618df9102
```

Comment:
https://github.com/Byron/crates-index-diff-rs/commit/8256cbbd3651073394d6aa9de38c734618df9102#diff-f4d390db9399b54cb9636f1c8e4bdb1822678936ea35fc93b537c9fb134c8a5cL35-L44

Reply:
I am not as well versed in the details of `docs.rs` but it seems that a build should always be queued for `AddedAndYanked`. From what I understood the `yank` will always fail anyway because the release is not yet know but the crate should be atleast be added to the build queue. Therefore if I had to choose I  would place `AddedAndYanked` in the `Added` mach arm instead of the `Yanked` match arm. However I guess the failure to yank should also be logged (or ideally succeed in the future)

However I added this variant I intended dowstream crates to use on the `added` and `yanked` function instead of matching on the variants directly. These correctly account for the fact that `AddedAndYanked` corresponds to both an `Added` and a `Yanked` event. Therefore I would suggest something like the following:

``` rust
 if Some(deleted) = change.deleted(){
// Match arm for Changed::Deleted
// deleted crates can not be yanked, continue
continue
}

// yanked() includes both Yanked and AddedAndYanked
if let Some(added) = change.added(){
// Match arm for Changed::Added
}


// yanked() includes both Yanked and AddedAndYanked
let yanked = changed.yanked();
let unyanked = changed.unyanked();
if let Some(release) = yanked.or(unyanked){
    // FIXME: delay yanks of crates that have not yet finished building
    set_yanked(
        &mut conn,
        release.name.as_str(),
        release.version.as_str(),
        yanked.is_some(),
    );

    if let Err(err) = cdn::invalidate_crate(&self.config, &self.cdn, &release.name)
    {
        report_error(&err);
    }
};
```

Reply:
>  Therefore if I had to choose I would place AddedAndYanked in the Added mach arm instead of the Yanked match arm. However I guess the failure to yank should also be logged (or ideally succeed in the future)

Thinking longer about this, you're right, this makes more sense.

Reply:
I adapted our code to use the new methods